executable-spec/test/Cardano/Ledger/Spec/STS/Chain/Chain/Properties.hs

```diff
@@ -46,3 +53,243 @@ qc_revealsAreClassified
       & dom
       & length
       & fromIntegral
+
+-- | Test coverage describes the test space that we are targeting for.
+-- Describe the most essential characteristics of our generated traces.
+--
+-- TODO: this must be updated to reflect all phases in the lifecycle of
+-- a software update and not just the Ideation phase.
+relevantCasesAreCovered :: QC.Property
+relevantCasesAreCovered
+  = QC.withMaxSuccess 300
+  $ STS.Gen.forAllTrace @(CHAIN ShortHash) @() maxTraceLength ()
+  $ \traceSample ->
+      -- traces should be long enough to allow for the stabilization
+      -- of events
+      QC.cover 80
+        ( (fromIntegral $ Trace.traceLength traceSample)
+          >=
+          getMinTraceLength  (Chain.k $ Trace._traceEnv traceSample)
+        )
+        "Trace length is long enough to allow for the stabilization of events"
+        $
+
+      -- 80% of traces should include a 20% percent of update payload
+      QC.cover 80
+        ( (updatePayloadPct traceSample) >= 15
+        &&
+          (updatePayloadPct traceSample) <= 25
+        )
+        "a reasonable pct of update payload appears in the trace "
+        $
+
+      -- Lifecycle coverage:
+      -- There is at least one proposal in every phase of the lifecycle
+      QC.cover 50
+        (lifecycleCoverage traceSample)
```

Comment:
I'm not sure about the value of this check. We're checking that there are SIP submissions, revelations, votes on SIP's, etc. So the checks below seem to subsume this one. 

Is this telling us something more?

Reply:
We are checking them individually and not with an AND logic

Reply:
The lifecycle check includes more than just the existence of submission, reveals and ballots. It also includes the check for active sips in the state and for approved SIPs. It is an overall check for the lifecycle. However, there is indeed an overlap. But I suggest we keep it.